cli/fathom_web/commands/vectorize.py

```diff
@@ -0,0 +1,312 @@
+                error = extract_error_from(status_box.text)
+                print(f'Vectorization failed with error:\n{error}')
+                break
+            now_completed_samples = len([line for line in status_box.text.splitlines() if line.endswith(': vectorized')])
+            bar.update(now_completed_samples - completed_samples)
+            completed_samples = now_completed_samples
+
+    new_file = look_for_new_vector_file(downloads_dir, vector_files_before)
+    print(f'Vectors saved to {str(new_file)}')
+    return firefox
+
+
+def get_fathom_fox_uuid(firefox):
+    """
+    Try to get the internal UUID for FathomFox from `prefs.js`.
+
+    We use a loop to try multiple times because the `prefs.js` file needs a
+    little time to update before the fathom addon information appears. Five
+    seconds seems adequate since one second has always worked for me (Daniel).
+    """
+    for _ in range(5):
+        prefs = (pathlib.Path(firefox.capabilities.get('moz:profile')) / 'prefs.js').read_text().split(';')
+        uuids = next((line for line in prefs if 'extensions.webextensions.uuids' in line)).split(',')
+        try:
+            fathom_fox_uuid = next((line for line in uuids if 'fathomfox@mozilla.com' in line)).split('\\"')[3]
+            return fathom_fox_uuid
+        except StopIteration:
+            time.sleep(1)
+    else:
+        raise GracefulError('Could not find UUID for FathomFox. No entry in the pres.js file.')
+
+
+def vector_files_present(firefox):
+    """Return the number of vector-y files present in Firefox's download dir."""
+    download_dir = pathlib.Path(firefox.profile.default_preferences['browser.download.dir'])
+    vector_files = download_dir.glob('vector*.json')
+    return len(list(vector_files))
+
+
+def extract_error_from(status_text):
+    """
+    Look for errors in the vectorizer's status text.
+
+    If there is an error, raise an exception causing an ungraceful shutdown.
+    """
+    lines = status_text.splitlines()
+    for line in lines:
+        if 'failed:' in line:
+            return line
+    raise UngracefulError(f'There was a vectorizer error, but we could not find it in {status_text}')
+
+
+def look_for_new_vector_file(downloads_dir, vector_files_before):
+    """
+    Look for a new vector file in the downloads directory.
+
+    We use a loop to try multiple times because the file system needs a little
+    little time to update before the file appears. Five seconds seems adequate
+    since one second has always worked for me (Daniel).
+    """
+    for _ in range(5):
+        vector_files_after = set(downloads_dir.glob('vector*.json'))
+        try:
+            new_file = (vector_files_after - vector_files_before).pop()
+            return new_file
+        except KeyError:
+            time.sleep(1)
+    else:
+        raise GracefulError(f'Could not find vectors file in {downloads_dir.as_posix()}.\n{vector_files_before} were already present.')
```

Comment:
> Firstly, since the error occurs after the vectorization has been attempted, that isn't an error during set up, is it? Maybe we should precede the error with "Error during setup" in this case.

I'll change the handling of `GracefulError` to remove reference to setup.
 
> Also, if we print an error that one of the pages failed during vectorization , do we need to still check and throw the `GracefulError` also? We probably shouldn't `look_for_new_vector_file` if the vectorizer throws an error, since it stops vectorizing in this case anyway.

Ah, I don't know how I didn't catch this one. We should be throwing a `UngracefulError` here because this is an error during vectorization and I haven't found a way to close everything during vectorization.

> Finally, in my case of first running this, `vector_files_before` is an empty set, so it just prints `set()` in the error message. Maybe if the set is empty, we just log that no vectors files were found in the download directory?

That is a good suggestion.